spring-cloud-gcp-docs/src/main/asciidoc/pubsub.adoc

```diff
@@ -0,0 +1,271 @@
+}
+----
+
+A Spring Boot starter is provided to auto-configure the various Pub/Sub components.
+
+Maven coordinates, using Spring Cloud GCP BOM:
+
+[source,xml]
+----
+<dependency>
+    <groupId>org.springframework.cloud</groupId>
+    <artifactId>spring-cloud-gcp-starter-pubsub</artifactId>
+</dependency>
+----
+
+Gradle coordinates:
+
+[source]
+----
+dependencies {
+    compile group: 'org.springframework.cloud', name: 'spring-cloud-gcp-starter-pubsub', version: '1.0.0.BUILD-SNAPSHOT'
+}
+----
+
+=== Pub/Sub operations abstraction
+
+`PubSubTemplate` is an abstraction that allows Spring users to use Google Cloud Pub/Sub without
+depending on any Google Cloud Pub/Sub API semantics. `PubSubTemplate` uses the
+https://github.com/GoogleCloudPlatform/google-cloud-java/tree/master/google-cloud-pubsub[Google Cloud Java Client for Pub/Sub]
+to interact with Google Cloud Pub/Sub. It implements the `PubSubOperations` interface, which
+comprehends all the possible operations to be carried by a Google Cloud Pub/Sub client.
+
+`PubSubTemplate` depends on a `PublisherFactory`, which is a functional interface to provide a
+Google Cloud Java Client for Pub/Sub `Publisher`. The Spring Boot starter for GCP Pub/Sub
+auto-configures a `PublisherFactory` with default settings and uses the `GcpProjectIdProvider` and
+`CredentialsProvider` auto-configured by the Spring Boot GCP Core starter.
+
+The `PublisherFactory` implementation provided by Spring Cloud GCP Pub/Sub,
+`DefaultPublisherFactory`, caches `Publisher` instances by topic name, in order to optimize resource
+utilization.
+
+The Spring Boot starter for GCP Pub/Sub allows for the customization of the number of threads used
+by both the `Publisher` and `Subscriber` executors. In the example, we're setting the number of
+threads used by both the `Publisher` and `Subscriber` to 2.
+
+[source,yaml]
+----
+spring.cloud.gcp.pubsub.subscriber.executorThreads=2
+spring.cloud.gcp.pubsub.publisher.executorThreads=2
+----
+
+The default value for both these properties is 4 threads.
+
+==== Publishing to a topic
+
+`PubSubTemplate` provides an asynchronous method to publish messages to a Google Cloud Pub/Sub
+topic.
+
+[source,java]
+----
+public ListenableFuture<String> send(final String topic, Message message)
+----
+
+The `send()` method takes a Spring Message and it uses the built-in `SimpleMessageConverter` to
+convert from `Message` to the `PubsubMessage` accepted by the Google Cloud Java Library for Pub/Sub.
+`SimpleMessageConverter` copies the `Message` headers into `PubsubMessage` attributes and the
+`Message` payload into the `PubsubMessage` data. `SimpleMessageConverter` calls `toString()` in both
+the `Message` headers and payload, since `PubsubMessage` only supports `String` attributes and data
+at the transmission level. We're considering JSON object encoding/decoding in the future to support
```

Comment:
I wouldn't say anything about future plans.
First: plans might be changes.
Second: we may forget to remove this sentence when the fix is there already.

OTOH do we have an appropriate GH issue if plans are announced?
😄

Reply:
I meant to address this object conversion issue in two places: #54 and then #92.
Because this is such a relevant issue, I'm pretty positive we will change the text when we get around to implementing this!